### PR TITLE
Stop tftp service when xcatd shutdown completes

### PR DESCRIPTION
**Pull request by chenglch**
close-issue: #2436

### CODE REVIEW

File: xCAT-server/lib/xcat/plugins/AAsn.pm [hunk 1]
```diff
@@ -1262,29 +1266,95 @@ sub setup_HTTP
...
+#-----------------------------------------------------------------------------
+#
+#=head3 enable_TFTPhpa
+#
+#    Configure and enable the tftp-hpa in the xinetd.
+#
+#=cut
+#
+#-----------------------------------------------------------------------------
+sub enable_TFTPhpa
```
Comment (neo954) [inline xCAT-server/lib/xcat/plugins/AAsn.pm]: On Ubuntu, `tftpd-hpa` is another tftp server. It is "HPA's tftp server". Package `tftpd-hpa` conflict with `tftpd`, but provide similar function. The end user can choose to use `tftpd-hpa` to replace the ordinary `tftpd`. It is just another implementation of the tftp server.

I believe the function `enable_TFTPhpa` here is used to handle the `tftpd-hpa` in Ubuntu. Thus, change this function massively may change its original behavior. I think this change is improper.

File: xCAT-server/sbin/xcatd [hunk 2]
```diff
@@ -1549,6 +1549,23 @@ if ($dbmaster) {
 
 # stop the monitoring process
 xCAT_monitoring::monitorctrl::stop($$);
+stop_tftp_service();
```
Comment (immarvin) [inline xCAT-server/sbin/xcatd]: will tftp process be stopped while stopping xcatd with SIGINT or SIGTERM? when xcatd is started with "xcatd -f", it is usually terminated with ``CTRL+C``

File: xCAT-server/lib/xcat/plugins/AAsn.pm (hunk 1 above)
Comment (immarvin) [inline xCAT-server/lib/xcat/plugins/AAsn.pm]: hi @neo954 , the subroutine name might be misleading here, in the body of the subroutine, the ordinary tftp, tftpd-hpa and atftp are processed.

File: xCAT-server/sbin/xcatd (hunk 2 above)
Comment (chenglch) [inline xCAT-server/sbin/xcatd]: yes, it will be stopped.

File: xCAT-server/lib/xcat/plugins/AAsn.pm (hunk 1 above)
Comment (chenglch) [inline xCAT-server/lib/xcat/plugins/AAsn.pm]: This is a legacy problem, I will use `enable_TFTP` instead

**Comment by immarvin** [comment]
agree to merge

**Pull request merged**